source/slang/slang-ir-wgsl-legalize.cpp

```diff
@@ -674,6 +675,13 @@ struct LegalizeWGSLEntryPointContext
             auto newKey = builder.createStructKey();
             copyNameHintAndDebugDecorations(newKey, oldKey);
 
+            if (auto interpolationModeDecoration =
```

Comment:
should we transfer all decorations from oldKey to newKey?

Reply:
Seems to work for the `nointerpolation` test, at least.
Should we do the same thing for the field, and not just the key?